projects/packages/blaze/src/class-dashboard-rest-controller.php

```
@@ -329,6 +351,30 @@ public function get_dsp_blaze_posts( $req ) {
 		return $this->get_dsp_generic( sprintf( 'v1/wpcom/sites/%d/blaze/posts', $site_id ), $req );
 	}
 
+	/**
+	 * Redirect GET requests to WordAds DSP Blaze media endpoint for the site.
+	 *
+	 * @param WP_REST_Request $req The request object.
+	 * @return array|WP_Error
+	 */
+	public function get_dsp_media( $req ) {
+		$site_id = $this->get_site_id();
+		if ( is_wp_error( $site_id ) ) {
+			return array();
```

I played a bit with the code and started thinking what if we, using the the Fail Fast approach, return `$site_id` if it's an instance of `WP_Error`?

I agree that sometimes we return an empty array but I do believe that it would be better to return an error instead.
For example the function `request_as_user` that's used by `get_dsp_generic`, we use here, actually returns an error is something went wrong with the request ([here](https://github.com/Automattic/jetpack/blob/trunk/projects/packages/blaze/src/class-dashboard-rest-controller.php#L587)).

Also, I checked that returning `WP_Error` instead of an empty array will produce `500` error but it is handled fine.

---

ignoring for now